Source/WebCore/platform/gstreamer/GStreamerQuirks.h

```diff
@@ -44,7 +46,18 @@ class GStreamerQuirkBase {
     GStreamerQuirkBase() = default;
     virtual ~GStreamerQuirkBase() = default;
 
-    virtual const char* identifier() = 0;
+    virtual const ASCIILiteral identifier() const = 0;
+
+    // Interface of classes supplied to MediaPlayerPrivateGStreamer to store values that the quirks will need for their job.
+    class GStreamerQuirkState {
+        WTF_MAKE_FAST_ALLOCATED;
```

Comment:
Maybe making it non-copyable?

Reply:
Wow! This change unveiled a lot of hidden copies that were happening under the hood. I had to define a move constructor ( `= default`, but anyway) both in this base class and in the GStreamerQuirkBroadcomBase subclass, as well as declare some `auto&` instead of `auto` in a couple of places.

I also found out that `makeUnique()` requires a copy, so I had to construct the `std::unique_ptr`s by doing `std::unique_ptr<GStreamerQuirkBase::GStreamerQuirkState>(new GStreamerQuirkBase::GStreamerQuirkState(WTFMove(state)))` in `setQuirkState()` instead. C++ doesn't lose an opportunity to shoot coders in the foot, haha. 🤭

Reply:
The checks that I added (in a debug branch) after adding the move constructor unveiled an even worse problem: [object slicing](https://en.wikipedia.org/wiki/Object_slicing) 😱
On superclass/subclass hierarchies, it's a really bad idea to move (or to copy) an instance of the subclass casted as the superclass, since the specific data of the subclass is **lost** during the move/copy. To prevent this to myself and to future developers, I've declared GStreamerQuirkState as non-copyable (as you said) and also non-movable.

As you suggested to me in private, the right solution is to modify `setQuirkState()` to take pre-made `unique_ptr`, which would be created by the caller (the subclass code, in this case) and would hold an instance of the right class. That instance would never be moved/copied and would therefore not suffer from slicing. And if it did (by some developer mistake in the future), the `WTF_MAKE_NONCOPYABLE` and `WTF_MAKE_NONMOVABLE` macros (which actually `= delete` the copy/move constructors).